### ISSUE TITLE
Type not assignable on CSS Prop when using on styled components

### ISSUE THREAD

**Issue by Henrixounez** [jantimon/next-yak#241]
When using the CSS Prop on a styled component to override the style, there is a Typescript error complaining about the type being assigned.
Adding the `jsxImportSource` on top of the file or in `tsconfig` doesn't solve this problem.

👉 The dev mode works correctly and renders the component with the correct style applied, but it won't pass the typecheck when building.

## ✅ Working when using on vanilla HTML Tags
```tsx
  <p
    css={css`
      color: red;
    `}
  >
    example page
  </p>
```

## ❌ Error when using on styled components
```tsx
/** @jsxImportSource next-yak */
import { styled, css } from "next-yak";

const SubHeadline = styled.h2`
  color: #323248;
`;

export default function Home() {
  return (
    <SubHeadline
      css={css`
        color: green;
      `}
    >
      example page
    </SubHeadline>
  );
}
```

![image](https://github.com/user-attachments/assets/749957cc-7471-4f0c-8de4-3c93cfd57961)

```
$ npm run build

> next build

   ▲ Next.js 15.1.0
   - Experiments (use with caution):
     · swcPlugins

   Creating an optimized production build ...
 ✓ Compiled successfully
   Linting and checking validity of types  ..Failed to compile.

./app/page.tsx:128:11
Type error: Type '{ children: string; css: StaticCSSProp; }' is not assignable to type 'IntrinsicAttributes & ClassAttributes<HTMLHeadingElement> & HTMLAttributes<HTMLHeadingElement>'.
  Property 'css' does not exist on type 'IntrinsicAttributes & ClassAttributes<HTMLHeadingElement> & HTMLAttributes<HTMLHeadingElement>'.

  126 |         <Headline $primary>Hello world</Headline>
  127 |         <SubHeadline
> 128 |           css={css`
      |           ^
  129 |             color: green;
  130 |           `}
  131 |         >
 ELIFECYCLE  Command failed with exit code 1.
```

**Comment by jantimon** [comment]
This is fixed in 4.0.2 - please let us know if the fix works for you

**Comment by Mad-Kat** [comment]
This is fixed with yak `4.0.2`

Can you confirm this @Henrixounez ?
